src/core/device.c

```diff
@@ -349,6 +361,29 @@ static int device_setup_unit(Manager *m, struct udev_device *dev, const char *pa
                         (void) device_add_udev_wants(u, dev);
         }
 
+        DEVICE(u)->bind_mounts = device_is_bound_by_mounts(dev);
```

Comment:
hmm, you have a point maybe, but in that case I think the caching should be hidden in device_is_bound_by_mounts()